libs/vkd3d/command.c

```diff
@@ -3330,6 +3335,14 @@ static void vk_access_and_stage_flags_from_d3d12_resource_state(const struct d3d
             case D3D12_RESOURCE_STATE_DEPTH_WRITE:
             case D3D12_RESOURCE_STATE_DEPTH_READ:
                 *stages |= VK_PIPELINE_STAGE_EARLY_FRAGMENT_TESTS_BIT | VK_PIPELINE_STAGE_LATE_FRAGMENT_TESTS_BIT;
+                *access |= VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_READ_BIT;
+                /* If our default DS layout is attachment optimal, we will not perform implicit
+                 * memory barriers as part of a render pass. */
+                if (state == D3D12_RESOURCE_STATE_DEPTH_WRITE &&
```

Comment:
nit: why not just move this above the `D3D12_RESOURCE_STATE_DEPTH_READ` thing and do a good old `/* fall through */` at the end?

Reply:
Fixed.